File: js_sdk/src/model/FilteringBcAccountTransactionGet.js
/*
 * Copyright 2023 TikTok Pte. Ltd.
 *
 * This source code is licensed under the MIT license found in
 * the LICENSE file in the root directory of this source tree.
 */
import {ApiClient} from '../ApiClient.js';

/**
 * The FilteringBcAccountTransactionGet model module.
 * @module model/FilteringBcAccountTransactionGet
 * @version 0.1.4
 */
export class FilteringBcAccountTransactionGet {
  /**
   * Constructs a new <code>FilteringBcAccountTransactionGet</code>.
   * @alias module:model/FilteringBcAccountTransactionGet
   * @class
   */
  constructor() {
  }

  /**
   * Constructs a <code>FilteringBcAccountTransactionGet</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/FilteringBcAccountTransactionGet} obj Optional instance to populate.
   * @return {module:model/FilteringBcAccountTransactionGet} The populated <code>FilteringBcAccountTransactionGet</code> instance.
   */
  static constructFromObject(data, obj) {
    if (data) {
      obj = obj || new FilteringBcAccountTransactionGet();
      if (data.hasOwnProperty('account_ids'))
        obj.account_ids = ApiClient.convertToType(data['account_ids'], ['String']);
      if (data.hasOwnProperty('account_name'))
        obj.account_name = ApiClient.convertToType(data['account_name'], 'String');
      if (data.hasOwnProperty('billing_types'))
        obj.billing_types = ApiClient.convertToType(data['billing_types'], ['String']);
      if (data.hasOwnProperty('end_time'))
        obj.end_time = ApiClient.convertToType(data['end_time'], 'String');
      if (data.hasOwnProperty('owner_of_account'))
        obj.owner_of_account = ApiClient.convertToType(data['owner_of_account'], ['String']);
      if (data.hasOwnProperty('start_time'))
        obj.start_time = ApiClient.convertToType(data['start_time'], 'String');
      if (data.hasOwnProperty('transaction_types'))
        obj.transaction_types = ApiClient.convertToType(data['transaction_types'], ['String']);
    }
    return obj;
  }
}

/**
 * List of account IDs.
 * @member {Array.<String>} account_ids
 */
FilteringBcAccountTransactionGet.prototype.account_ids = undefined;

/**
 * Account name.
 * @member {String} account_name
 */
FilteringBcAccountTransactionGet.prototype.account_name = undefined;

/**
 * Billing types.
 * @member {Array.<String>} billing_types
 */
FilteringBcAccountTransactionGet.prototype.billing_types = undefined;

/**
 * End time for filtering transactions.
 * @member {String} end_time
 */
FilteringBcAccountTransactionGet.prototype.end_time = undefined;

/**
 * Owners of the account.
 * @member {Array.<String>} owner_of_account
 */
FilteringBcAccountTransactionGet.prototype.owner_of_account = undefined;

/**
 * Start time for filtering transactions.
 * @member {String} start_time
 */
FilteringBcAccountTransactionGet.prototype.start_time = undefined;

/**
 * Transaction types.
 * @member {Array.<String>} transaction_types
 */
FilteringBcAccountTransactionGet.prototype.transaction_types = undefined;

File: java_sdk/src/main/java/business_api_client/FilteringBcAccountTransactionGet.java
/*
 * Copyright 2023 TikTok Pte. Ltd.
 *
 * This source code is licensed under the MIT license found in
 * the LICENSE file in the root directory of this source tree.
 */

package business_api_client;

import java.util.Objects;
import java.util.Arrays;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import io.swagger.v3.oas.annotations.media.Schema;
import java.util.ArrayList;
import java.util.List;
/**
 * FilteringBcAccountTransactionGet
 */

@javax.annotation.Generated(value = "com.tiktok.codegen.JavatiktokcodegenGenerator", date = "2025-07-03T14:36:18.961395-07:00[America/Los_Angeles]")
public class FilteringBcAccountTransactionGet {
  @JsonProperty("account_ids")
  private List<String> accountIds = null;

  @JsonProperty("account_name")
  private String accountName = null;

  @JsonProperty("billing_types")
  private List<String> billingTypes = null;

  @JsonProperty("end_time")
  private String endTime = null;

  @JsonProperty("owner_of_account")
  private List<String> ownerOfAccount = null;

  @JsonProperty("start_time")
  private String startTime = null;

  @JsonProperty("transaction_types")
  private List<String> transactionTypes = null;

  public FilteringBcAccountTransactionGet accountIds(List<String> accountIds) {
    this.accountIds = accountIds;
    return this;
  }

  public FilteringBcAccountTransactionGet addAccountIdsItem(String accountIdsItem) {
    if (this.accountIds == null) {
      this.accountIds = new ArrayList<String>();
    }
    this.accountIds.add(accountIdsItem);
    return this;
  }

   /**
   * List of account IDs.
   * @return accountIds
  **/
  @Schema(description = "List of account IDs.")
  public List<String> getAccountIds() {
    return accountIds;
  }

  public void setAccountIds(List<String> accountIds) {
    this.accountIds = accountIds;
  }

  public FilteringBcAccountTransactionGet accountName(String accountName) {
    this.accountName = accountName;
    return this;
  }

   /**
   * Account name.
   * @return accountName
  **/
  @Schema(description = "Account name.")
  public String getAccountName() {
    return accountName;
  }

  public void setAccountName(String accountName) {
    this.accountName = accountName;
  }

  public FilteringBcAccountTransactionGet billingTypes(List<String> billingTypes) {
    this.billingTypes = billingTypes;
    return this;
  }

  public FilteringBcAccountTransactionGet addBillingTypesItem(String billingTypesItem) {
    if (this.billingTypes == null) {
      this.billingTypes = new ArrayList<String>();
    }
    this.billingTypes.add(billingTypesItem);
    return this;
  }

   /**
   * Billing types.
   * @return billingTypes
  **/
  @Schema(description = "Billing types.")
  public List<String> getBillingTypes() {
    return billingTypes;
  }

  public void setBillingTypes(List<String> billingTypes) {
    this.billingTypes = billingTypes;
  }

  public FilteringBcAccountTransactionGet endTime(String endTime) {
    this.endTime = endTime;
    return this;
  }

   /**
   * End time for filtering transactions.
   * @return endTime
  **/
  @Schema(description = "End time for filtering transactions.")
  public String getEndTime() {
    return endTime;
  }

  public void setEndTime(String endTime) {
    this.endTime = endTime;
  }

  public FilteringBcAccountTransactionGet ownerOfAccount(List<String> ownerOfAccount) {
    this.ownerOfAccount = ownerOfAccount;
    return this;
  }

  public FilteringBcAccountTransactionGet addOwnerOfAccountItem(String ownerOfAccountItem) {
    if (this.ownerOfAccount == null) {
      this.ownerOfAccount = new ArrayList<String>();
    }
    this.ownerOfAccount.add(ownerOfAccountItem);
    return this;
  }

   /**
   * Owners of the account.
   * @return ownerOfAccount
  **/
  @Schema(description = "Owners of the account.")
  public List<String> getOwnerOfAccount() {
    return ownerOfAccount;
  }

  public void setOwnerOfAccount(List<String> ownerOfAccount) {
    this.ownerOfAccount = ownerOfAccount;
  }

  public FilteringBcAccountTransactionGet startTime(String startTime) {
    this.startTime = startTime;
    return this;
  }

   /**
   * Start time for filtering transactions.
   * @return startTime
  **/
  @Schema(description = "Start time for filtering transactions.")
  public String getStartTime() {
    return startTime;
  }

  public void setStartTime(String startTime) {
    this.startTime = startTime;
  }

  public FilteringBcAccountTransactionGet transactionTypes(List<String> transactionTypes) {
    this.transactionTypes = transactionTypes;
    return this;
  }

  public FilteringBcAccountTransactionGet addTransactionTypesItem(String transactionTypesItem) {
    if (this.transactionTypes == null) {
      this.transactionTypes = new ArrayList<String>();
    }
    this.transactionTypes.add(transactionTypesItem);
    return this;
  }

   /**
   * Transaction types.
   * @return transactionTypes
  **/
  @Schema(description = "Transaction types.")
  public List<String> getTransactionTypes() {
    return transactionTypes;
  }

  public void setTransactionTypes(List<String> transactionTypes) {
    this.transactionTypes = transactionTypes;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    FilteringBcAccountTransactionGet filteringBcAccountTransactionGet = (FilteringBcAccountTransactionGet) o;
    return Objects.equals(this.accountIds, filteringBcAccountTransactionGet.accountIds) &&
        Objects.equals(this.accountName, filteringBcAccountTransactionGet.accountName) &&
        Objects.equals(this.billingTypes, filteringBcAccountTransactionGet.billingTypes) &&
        Objects.equals(this.endTime, filteringBcAccountTransactionGet.endTime) &&
        Objects.equals(this.ownerOfAccount, filteringBcAccountTransactionGet.ownerOfAccount) &&
        Objects.equals(this.startTime, filteringBcAccountTransactionGet.startTime) &&
        Objects.equals(this.transactionTypes, filteringBcAccountTransactionGet.transactionTypes);
  }

  @Override
  public int hashCode() {
    return Objects.hash(accountIds, accountName, billingTypes, endTime, ownerOfAccount, startTime, transactionTypes);
  }


  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class FilteringBcAccountTransactionGet {\n");
    
    sb.append("    accountIds: ").append(toIndentedString(accountIds)).append("\n");
    sb.append("    accountName: ").append(toIndentedString(accountName)).append("\n");
    sb.append("    billingTypes: ").append(toIndentedString(billingTypes)).append("\n");
    sb.append("    endTime: ").append(toIndentedString(endTime)).append("\n");
    sb.append("    ownerOfAccount: ").append(toIndentedString(ownerOfAccount)).append("\n");
    sb.append("    startTime: ").append(toIndentedString(startTime)).append("\n");
    sb.append("    transactionTypes: ").append(toIndentedString(transactionTypes)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }

}
